File: python/7 kyu/268.py
# https://www.codewars.com/kata/5a34da5dee1aae516d00004a/train/python

# Passed

def get_matrix(n):
    matrix = []
    for idx in range(n):
        row = [0] * n
        row[idx] = 1
        matrix.append(row)
        
    return matrix

output = get_matrix(3)
print(output)

File: nodejs/7 kyu/268.js
// https://www.codewars.com/kata/56d6b7e43e8186c228000637/train/javascript

// Passed

function colourAssociation(array) {
  let colourToAttributeArr = array.map((item) => {
    let [colour, attribute] = item;
    let colourToAttributeObj = new Object();
    colourToAttributeObj[colour] = attribute;
    return colourToAttributeObj;
  });

  return colourToAttributeArr;
}

const output = colourAssociation([
  ["white", "goodness"],
  ["blue", "tranquility"],
]);
console.log(output);